[ACR-1374,1389,1393] Add new sync transform APIs and Update Image Decoding/Encoding APIs

jiyong2 opened the pull request:
- ACR: 
ACR-1374
[5.5][capi-media-image-util][Add] Add new sync transform APIs (convert, crop, resize, rotate)
ACR-1389
[5.5][capi-media-image-util][Add/Deprecate] Update Image Decoding APIs

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -194,38 +256,77 @@ To rotate an image:
+
+## Rotating a Media Packet
+
+To rotate a media_packet:
```
Global: Suggest rephrasing to:

To resize a media packet:

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
````diff
@@ -282,14 +421,7 @@ To decode a JPEG, PNG, GIF, or BMP image:
-   ```
-
-4. Optionally, set the color space and JPEG downscale using the `image_util_decode_set_colorspace()` and `image_util_decode_set_jpeg_downscale()` functions:
+3. Optionally, set the color space and JPEG downscale using the `image_util_decode_set_colorspace()` and `image_util_decode_set_jpeg_downscale()` functions:
````
Can we rephrase to:

3. Additionally, you can set the color space.....

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
````diff
@@ -329,31 +459,7 @@ To encode a raw image:
-   ```
-
-5. Optionally, set the JPEG quality or PNG compression using the `image_util_encode_set_quality()` or `image_util_encode_set_png_compression()` function:
+2. Optionally, set the JPEG quality or PNG compression using the `image_util_encode_set_quality()` or `image_util_encode_set_png_compression()` function:
````
Global: Can we rephrase to:

2. Additionally, you can set the ..............

jiyong2 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -88,62 +84,93 @@ To enable your application to use the image util functionality:
-   src = malloc(size);
-   ret = media_packet_get_buffer_data_ptr(src, &src_ptr);
-   ret = memcpy(src_ptr, img_source, size);
+1. Create a transformation handle using the `image_util_transform_create()` function:
```
Thanks for your review.
I will remove "the" and "function" in case of "the *** function".
I have one question. Should I remove "the" and "callback" in case of "the *** callback"?

jiyong2 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -88,62 +84,93 @@ To enable your application to use the image util functionality:
    > - Here, the `image_util_transform_run()` function only converts the color space. The function does not change the image width or height, or any other image property.
-   > - Because of the restrictions of the image processing library, not all color space combinations are supported for conversion. For example, the NV12 format is commonly used in hardware chips, but it is not supported by the library.
-   > - If hardware acceleration is enabled, you can execute 2 more image transformations using the same transformation handle.
+   > - Because of the restrictions of the image processing library, not all color space combinations are supported for conversion.
```
Thanks for your review.
[Media Handle Management] looks good. I will change  [Media Tool API] to [Media Handle Management].
If you have another idea, please let us know it.

jiyong2 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -154,29 +181,64 @@ To resize an image:
+
+## Resizing a Media Packet
+
+To resize a media_packet:
```
Thanks, I will remove the "_".

jiyong2 commented on docs/application/native/guides/multimedia/image-edit.md:
````diff
@@ -88,62 +84,93 @@ To enable your application to use the image util functionality:
+   ret = image_util_transform_destroy(transform_h);
+   ```
+
+## Converting the Color Space of a Media Packet
````
@mijin85cho Yes, I added some sections in this page. I modified simply introduces the features and added new links. Could you please check the modification?
- Conversion
  You can change the [image](#image_colorspace) and [media packet](#mediapacket_colorspace) color space among the [supported formats](#color_format).
- Resizing
  You can change the [image](#resize_image) and [media packet](#resize_mediapacket) resolution.
- Rotation
  You can change the [image](#rotate_image) and [media_packet](#rotate_mediapacket) angle around the X or Y axis.
- Crop
  You can remove the outer parts of the [image](#crop_image) and [media packet](#crop_mediapacket).

jiyong2 commented:
Thanks for your reviews.
I have updated the guide. Please check it again.
 - The articles was removed.
 - The bullet points was fixed to your comments.
 - The words "the" and "function" in case of "the *** function" was removed in this page.
 -  I modified simply introduces the features and added new links.

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -82,214 +78,348 @@ To enable your application to use the image util functionality:
+
+   > **Note**
+   >
+   > - `image_util_transform_run2()` and `image_util_transform_run2_async()` only converts the color space. These functions do not change the image width, height, or any other image property. Due to these restrictions of the image processing library, not all color space combinations are supported for conversion.
```
Global for Note: Suggest removing the bulleted tag (-) before the content.

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -82,214 +78,348 @@ To enable your application to use the image util functionality:
+   >
+   > - `image_util_transform_run2()` and `image_util_transform_run2_async()` only converts the color space. These functions do not change the image width, height, or any other image property. Due to these restrictions of the image processing library, not all color space combinations are supported for conversion.
+
+5. Handle the transformation results in the `image_util_transform_completed2_cb()` callback, which is invoked after the transformation is complete.
```
Global for callback: Suggest removing the and callback. Here in the `image_util_transform_completed2_cb(), cb denotes callback. Therefore, suggest rephrasing to:

 Handle the transformation results in `image_util_transform_completed2_cb()`, which is invoked after the transformation is complete.

jiyong2 replied:
okay, I will remove "the" and "callback" in case of "the *** callback"

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -82,214 +78,348 @@ To enable your application to use the image util functionality:
+## Converting Media Packet Color Space
 
-To convert an image from one color space to another:
+To convert an image from one color space of a media_packet to another:
```
Can we rephrase media_packet to media packet? I am not sure if we are referring to something specific here.

To convert an image from one color space of a media packet to another:

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -82,214 +78,348 @@ To enable your application to use the image util functionality:
-   > - Because of the restrictions of the image processing library, not all color space combinations are supported for conversion. For example, the NV12 format is commonly used in hardware chips, but it is not supported by the library.
-   > - If hardware acceleration is enabled, you can execute 2 more image transformations using the same transformation handle.
+   > - `image_util_transform_run()` only converts the color space. The function does not change the image width, height, or any other image property. Due to these restrictions of the image processing library, not all color space combinations are supported for conversion.
+   > - For more information on how to use the media packet handle of the Media Tool API, see the [Media Handle Management](media-handle.md).
```
Suggest removing the bulleted tag (-) before the content.

surbhig12 replied:
Please incorporate this comment. Also, please remove the word the before the link:

.... Tool API, see [Media Handle Management](media-handle.md).

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -82,214 +78,348 @@ To enable your application to use the image util functionality:
+   > - For more information on how to use the media packet handle of the Media Tool API, see the [Media Handle Management](media-handle.md).
 
-4. Handle the transformation results in the `image_util_transform_completed_cb()` callback, which is invoked after the transformation is complete.
+5. Handle the transformation results in the `image_util_transform_completed_cb()` callback, which is invoked after the transformation is complete.
```
Global: Suggest removing the and callback. Here in the `image_util_transform_completed_cb()`, cb denotes callback. Therefore, suggest rephrasing to:

Handle the transformation results in `image_util_transform_completed_cb()`, which is invoked after the transformation is complete.

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -82,214 +78,348 @@ To enable your application to use the image util functionality:
    > - The image format has no effect on the transformation.
-   > - If the color space is YUV, the target image width and height must be multiples of 8. This restriction does not apply to RGB images.
-   > - If hardware acceleration is enabled, you can execute 2 more image transformations using the same transformation handle.
+   > - If the color space is YUV, the width and height of the target image must be multiples of 8. This restriction does not apply to the RGB images.
```
Suggest removing the bulleted tag (-) before the content.

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -154,29 +181,64 @@ To resize an image:
+   > - The image format has no effect on the transformation.
+   > - If the color space is YUV, the target image width and height must be multiples of 8. This restriction does not apply to RGB images.
+
+4. If `image_util_transform_run2_async()` is used to run the transformation, handle the transformation results in the `image_util_transform_completed2_cb()` callback, which is invoked after the transformation is complete.
```
@mijin85cho, yes I agree with you.

@jiyong2 
Suggest rephrasing to:

4. If `image_util_transform_run2_async()` is used to run the transformation, handle the transformation results in `image_util_transform_completed2_cb()` , which is invoked after the transformation is complete.

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -82,214 +78,348 @@ To enable your application to use the image util functionality:
+
+   > **Note**
+   >
+   > - The image format has no effect on the transformation. If the color space is YUV, then the width and height of the target image must be multiples of 8. This restriction does not apply to the RGB images.
```
Suggest removing the bulleted tag (-) before the content.

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
````diff
@@ -194,38 +256,77 @@ To rotate an image:
+   ret = image_util_transform_run2(transform_h, src_image, &dst_image);
+   ```
+
+   > **Note**
````
Suggest removing the bulleted tag (-) before the content.

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -82,214 +78,348 @@ To enable your application to use the image util functionality:
    >
-   > - The image format has no effect on the transformation.
-   > - If the color space is YUV, the target image width and height must be multiples of 8. This restriction does not apply to RGB images.If hardware acceleration is enabled, you can execute 2 more image transformations using the same transformation handle.
+   > - The image format has no effect on the transformation. If the color space is YUV, then the width and height of the target image must be multiples of 8. This restriction does not apply to the RGB images.
```
Suggest removing the bulleted tag (-) before the content.

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -82,214 +78,348 @@ To enable your application to use the image util functionality:
+   >
+   > - The image format has no effect on the transformation. If the color space is YUV, then the width and height of the target image must be multiples of 8. This restriction does not apply to the RGB images.
 
 4. Handle the transformation results in the `image_util_transform_completed_cb()` callback, which is invoked after the transformation is complete.
```
Suggest removing the and callback. Here in the `image_util_transform_completed_cb()`, cb denotes callback. Therefore, suggest rephrasing to:
Handle the transformation results in `image_util_transform_completed_cb()` , which is invoked after the transformation is complete.

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -82,214 +78,348 @@ To enable your application to use the image util functionality:
+
+   > **Note**
+   >
+   > - The image format has no effect on the transformation. If the color space is YUV, then the width and height of the target image must be multiples of 8. This restriction does not apply to the RGB images.
```
Suggest removing the bulleted tag (-) before the content.

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -82,214 +78,348 @@ To enable your application to use the image util functionality:
 
    > **Note**
    >
    > - Because of a YUV restriction, and because the crop start position can be set arbitrarily, the cropped image width and height must be even.
```
Suggest removing the bulleted tag (-) before the content.

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -82,214 +78,348 @@ To enable your application to use the image util functionality:
-   > - If the color space is YUV, the target image width and height must be multiples of 8. This restriction does not apply to RGB images.If hardware acceleration is enabled, you can execute 2 more image transformations using the same transformation handle.
+   > - The image format has no effect on the transformation. If the color space is YUV, then the width and height of the target image must be multiples of 8. This restriction does not apply to the RGB images.
+
+4. Handle the transformation results in the `image_util_transform_completed2_cb()` callback, which is invoked after the transformation is complete.
```
Suggest removing the and callback. Here in the `image_util_transform_completed2_cb()`, cb denotes callback. Therefore, suggest rephrasing to:
Handle the transformation results in `image_util_transform_completed2_cb()`, which is invoked after the transformation is complete.

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -82,214 +78,348 @@ To enable your application to use the image util functionality:
+
+   > **Note**
+   >
+   > - The image format has no effect on the transformation. If the color space is YUV, then the width and height of the target image must be multiples of 8. This restriction does not apply to the RGB images.
```
Suggest removing the bulleted tag (-) before the content.

Also, suggest changing 8 to eight

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -321,111 +449,67 @@ To decode a JPEG, PNG, GIF, or BMP image:
-   > - The default color space is `IMAGE_UTIL_COLORSPACE_RGBA8888`. PNG, GIF and BMP images are encoded with `IMAGE_UTIL_COLORSPACE_RGBA8888`.
-
-4. To save the encoded image, set the output path or buffer using the `image_util_encode_set_output_path()` or `image_util_encode_set_output_buffer()` function:
+   > - Due to the encoder limitations, the quality setting is only supported for encoding the JPEG images, and compression is only supported for the PNG images. The default JPEG quality is 75. The default PNG compression is `IMAGE_UTIL_PNG_COMPRESSION_6`.
```
Suggest removing the bulleted tag (-) before the content.

jiyong2 replied:
okay, I will remove the tag (-) in the notes.

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -302,16 +432,14 @@ To decode a JPEG, PNG, GIF, or BMP image:
    > - Because of decoder limitations, color space setting and JPEG downscaling are only supported for JPEG images.
    > - The default color space is `IMAGE_UTIL_COLORSPACE_RGBA8888`. PNG, GIF and BMP images are decoded to `IMAGE_UTIL_COLORSPACE_RGBA8888`.
 
```
Suggest removing the bulleted tag (-) before the content.

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
````diff
@@ -321,111 +449,67 @@ To decode a JPEG, PNG, GIF, or BMP image:
-   unsigned long long size = 0; /* Encoded image size */
-   ret = image_util_encode_run(encode_h, &size);
-   ```
+   > - Due to the encoder limitations, the color space setting is only supported for encoding the JPEG images. The default color space is `IMAGE_UTIL_COLORSPACE_RGBA8888`. PNG, GIF and BMP images are encoded with `IMAGE_UTIL_COLORSPACE_RGBA8888`.
````
Suggest removing the bulleted tag (-) before the content.

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -7,19 +7,19 @@ The main features of the Image Util API include:
-  You can [remove the outer parts of an image](#crop) or change the aspect ratio.
+  You can remove the outer parts of the [image](#crop_image) and [media packet](#crop_mediapacket).
 
 - Decoding from a file or memory and encoding to a file or memory
```
Just  a suggestion:

Can we avouid using sub bullet for the Bitmap formts, Input image formats for decoding, and Output image formats for encoding?

Suggest rephrasing to:
- Bitmap formats: YUV420, YUV422, RGB888, RGBA8888, BGRA8888, ARGB8888
- Input image formats for decoding: JPEG, PNG, GIF, BMP
- Output image formats for encoding: JPEG, PNG, GIF, BMP, animated GIF

jiyong2 replied:
Is the note unnecessary?

surbhig12 replied:
@jiyong2, sorry for the miss, please remove the periods at the end of the sentences:

Suggest rephrasing to:

Bitmap formats: YUV420, YUV422, RGB888, RGBA8888, BGRA8888, and ARGB8888
Input image formats for decoding: JPEG, PNG, GIF, and BMP
Output image formats for encoding: JPEG, PNG, GIF, BMP, and animated GIF

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -82,214 +78,348 @@ To enable your application to use the image util functionality:
+
+   > **Note**
+   >
+   > - The image format has no effect on the transformation. If the color space is YUV, then the width and height of the target image must be multiples of 8. This restriction does not apply to the RGB images.
```
Suggest changing 8 to eight.

jiyong2 replied:
Ah, okay. I will change 8 to eight.

jiyong2 commented:
I have updated the guide. Please check it again.

- Some bullet points were fixed then bullet tags were removed.
- The words "the" and "callback" in case of "the *** callback" was removed in this page.
- 8 to eight
- The typos like media_packet were fixed.

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -82,214 +72,347 @@ To enable your application to use the image util functionality:
    >
-   > - Because of a YUV restriction, and because the crop start position can be set arbitrarily, the cropped image width and height must be even.
-   > - If hardware acceleration is enabled, you can execute 2 more image transformations using the same transformation handle.
+   > Because of a YUV restriction, and because the crop start position can be set arbitrarily, the cropped image width and height must be even.
```
This sentence seems to be a little wordy, can we rephrase it:

As there is a YUV restriction and the crop start position can be set arbitrarily, the cropped image width and height must be even. 

Please verify.

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
````diff
@@ -321,111 +441,67 @@ To decode a JPEG, PNG, GIF, or BMP image:
-   unsigned long long size = 0; /* Encoded image size */
-   ret = image_util_encode_run(encode_h, &size);
-   ```
+   > Due to the encoder limitations, the color space setting is only supported for encoding the JPEG images. The default color space is `IMAGE_UTIL_COLORSPACE_RGBA8888`. PNG, GIF and BMP images are encoded with `IMAGE_UTIL_COLORSPACE_RGBA8888`.
````
Global: Please add a serial comma before BMP

PNG, GIF, and BMP images.....

jiyong2 commented:
> Thank you for incorporating the comments. I have reviewed the file and provided minor comments. Please verify.

jiyong2 commented:
I'm sorry for my mistake..
Thanks for your review.
I updated the document for your minor comments.
Please review the document.

surbhig12 commented:
@jiyong2,
Thank you for incorporating the comments. Request you to please incorporate the following two comments that were already mentioned: 

Line number: 28-30: For the Bitmap formts, Input image formats for decoding, and Output image formats for encoding, please make it a sub-bulleted list. 
Line number: 152: In the Note, please remove the bullets.

jiyong2 commented:
Thanks for your comments too. I updated the page for your two comments.
Please check and review it.

surbhig12 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -7,39 +7,27 @@ The main features of the Image Util API include:
+  - Input image formats for decoding: JPEG, PNG, GIF, and BMP.
+  - Output image formats for encoding: JPEG, PNG, GIF, BMP, and [animated GIF](#animation).
 
     Pay attention to how the [image quality depends on the size](#quality) and compression ratio.
```
This seems to be a note. Also, this sentenceis applicable for all the above 3 sub bulleted list. Request you to please left align it.

> **Note**
>
> Pay attention to how the [image quality depends on the size](#quality) and compression ratio.

jiyong2 replied:
Hello, the sentence applicable for "decoding from a file *** memory".
I will left align it.

surbhig12 replied:
@jiyong2, please remove the periods at the end of the sentence as it is not a complete sentence.

 - Bitmap formats: YUV420, YUV422, RGB888, RGBA8888, BGRA8888, and ARGB8888
  - Input image formats for decoding: JPEG, PNG, GIF, and BMP
  - Output image formats for encoding: JPEG, PNG, GIF, BMP, and [animated GIF](#animation)

journey2w commented on docs/application/native/guides/multimedia/image-edit.md:
````diff
@@ -234,36 +335,74 @@ To crop an image:
+                                user_data);
+   ```
+
+   > **Note**
````
@surbhig12 
"If the color space is YUV, then the width and height of the target image must be multiples of eight."
Why did you add "then"? Did you want to emphasis the dependency between main clause and subordinate clause.

We usually write "if..., ..." without "then" except mandatory case.

You had better NOT change when there is no problem in a sentence.

surbhig12 replied:
@journey2w,
The intention to use "then" in this particular case was to stress and emphasize on the dependency. 
Yes, I agree that we usually write "if..., ..." without "then" except mandatory case.
However, this is a conditional clause and we use “then” if we need to stress. Since it was an important information, I used "then".
I referred to: https://english.stackexchange. com/questions/247323/dropping-then-in-if-statements.

Although, if you feel "then" is not required here. I would go with your suggestion.
Thank you for the explanation.

jiyong2 commented:
The sentence was updated. Please check it.

jiyong2 commented:
@surbhig12 Always thanks for your review. I removed the periods at the end of the sentences.
Please check it.

mijin85cho commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -82,214 +68,347 @@ To enable your application to use the image util functionality:
-   ret = image_util_transform_run(handle, src,
-                                  (image_util_transform_completed_cb)completed_callback,
-                                  user_data);
+   ret = image_util_transform_run2_async(transform_h, src_image,
```
`image_util_transform_run()` in the description, `image_util_transform_run2_async` in the sample code.

I think `image_util_transform_run2_async` would be correct as the functions with 2 in the name are used for Image. However, would you please check?

jiyong2 replied:
`image_util_transform_run()` in the description will be replaced to `mage_util_transform_run2()` or `mage_util_transform_run2_async()`.

mijin85cho commented on docs/application/native/guides/multimedia/image-edit.md:
````diff
@@ -321,111 +437,67 @@ To decode a JPEG, PNG, GIF, or BMP image:
    ```
-   ret = image_util_encode_set_resolution(encode_h, width, height);
-   ret = image_util_encode_set_colorspace(encode_h, IMAGE_UTIL_COLORSPACE_RGBA8888);
+   ret = image_util_encode_set_jpeg_downscale(decode_h, IMAGE_UTIL_DOWNSCALE_1_1);
````
Would you please check `image_util_encode_set_jpeg_downscale`? I can't find it in 5.5 native API.

jiyong2 replied:
`image_util_encode_set_jpeg_downscale` should be changed to `image_util_encode_set_quality` in the sample code.
I will replace the sample code.

mijin85cho commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -299,19 +418,16 @@ To decode a JPEG, PNG, GIF, or BMP image:
    >
-   > - Because of decoder limitations, color space setting and JPEG downscaling are only supported for JPEG images.
-   > - The default color space is `IMAGE_UTIL_COLORSPACE_RGBA8888`. PNG, GIF and BMP images are decoded to `IMAGE_UTIL_COLORSPACE_RGBA8888`.
+   > Due to the encoder limitations, the color space setting is only supported for encoding the JPEG images. The default color space is `IMAGE_UTIL_COLORSPACE_RGBA8888`. PNG, GIF, and BMP images are encoded with `IMAGE_UTIL_COLORSPACE_RGBA8888`.
```
Just for check. Is this note really talking about encoding limitation while the note itself is located in decoding section?
Moreover the same note is located in encoding section, too.

jiyong2 replied:
It's my mistake, it should be decoder limitation. Iit has same limitation for decoding and encoding the image.
I will replace the sentence.

mijin85cho commented on docs/application/native/guides/multimedia/image-edit.md:
````diff
@@ -321,111 +437,67 @@ To decode a JPEG, PNG, GIF, or BMP image:
 
    ```
-   ret = image_util_encode_set_jpeg_downscale(decode_h, IMAGE_UTIL_DOWNSCALE_1_1);
+   ret = image_util_encode_run(encode_h, decoded_image, file);
````
`image_util_encode_run_to_file()` or `image_util_encode_run_to_buffer()` in description, and the sample code is `image_util_encode_run()`.
I can see `image_util_encode_run()` in the API file anyway, however would you please check?

jiyong2 replied:
`image_util_encode_run` should be changed to `image_util_encode_run_to_file in the sample code.
I will replace the sample code.

mijin85cho commented on docs/application/native/guides/multimedia/image-edit.md:
````diff
@@ -321,111 +437,67 @@ To decode a JPEG, PNG, GIF, or BMP image:
 
    ```
-   ret = image_util_encode_destroy(encode_h);
+   ret = image_util_agif_encode_destroy(agif_encode_h);
````
`image_util_encode_destroy()` in description, `image_util_agif_encode_destroy()` in sample code.
Would you please check?

jiyong2 replied:
okay, `image_util_encode_destroy()` should be changed to `image_util_agif_encode_destroy()` in description. I will replace it.

surbhig12 commented:
@jiyong2,
Request you to please incorporate @mijin85cho suggestions.

mijin85cho commented:
@jiyong2 Hello.
Would you please checked the additional comments and apply the fix? I also hope you to rebase to resolve the conflict.

jiyong2 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -82,214 +68,347 @@ To enable your application to use the image util functionality:
 
-   ret = media_packet_create_alloc(fmt, NULL, NULL, &src);
-   ret = media_packet_get_buffer_size(src, &size);
+To convert an image from one color space of an image to another:
```
Hello. Thanks for your review.
I agree the duplicated word and the meaning of sentence was wrong.
Because the feature is converting the colorspace of an image, I think that the sentence should be below. @all, Could you please check?

To convert one color space of an image to another:

mijin85cho replied:
@surbhig12 Would please check the suggestions? I think both from @jiyong2 are good.

In my opinion, 
If we are going to use "To convert one color space of a media packet to another:" for line 129, "To convert one color space of an image to another:" for here would be better.

surbhig12 replied:
@mijin85cho,

Thank you. I too agree with your point:
If we are going to use "To convert one color space of a media packet to another:" for line 129, "To convert one color space of an image to another:" for here would be better.

jiyong2 commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -82,214 +68,347 @@ To enable your application to use the image util functionality:
+<a name="mediapacket_colorspace"></a>
+## Converting Media Packet Color Space
+
+To convert an image from one color space of a media packet to another:
```
Because the feature is converting the colorspace of an image, I think that the sentence should be below.
Could you please check?

To convert one color space of a media packet to another:
To convert one color space of a media packet to another (color space)

surbhig12 replied:
@jiyong2, thank you for the response. As per me, 
To convert one color space of a media packet to another: 
is better.

jiyong2 commented:
The mismatch between description and sample code are updated.
Please check the guide again.

mijin85cho commented:
Thanks for the update, @jiyong2 
However I can see the image-edit.md has conflict. Would you please resolve it, also?

jiyong2 commented:
OK, I will check the conflicts.

jiyong2 commented:
The conflict was fixed.

mijin85cho commented:
Thanks for the update @jiyong2 
@surbhig12 , would you please re-checked the latest and approve it again?

mijin85cho commented on docs/application/native/guides/multimedia/image-edit.md:
```diff
@@ -300,19 +418,16 @@ To decode a JPEG, PNG, GIF, or BMP image:
    >
-   > - Because of decoder limitations, color space setting and JPEG downscaling are only supported for JPEG images.
-   > - The default color space is `IMAGE_UTIL_COLORSPACE_RGBA8888`. PNG, GIF and BMP images are decoded to `IMAGE_UTIL_COLORSPACE_RGBA8888`.
+   > Due to the decoder limitations, the color space setting is only supported for decoding the JPEG images. The default color space is `IMAGE_UTIL_COLORSPACE_RGBA8888`. PNG, GIF, and BMP images are encoded with `IMAGE_UTIL_COLORSPACE_RGBA8888`.
```
... PNG, GIF, and BMP images are **encoded** with...

According to the note (and @jiyong2's comment) as PNG, GIF and BMP are all encoded/decoded with IMAGE_UTIL_COLORSPACE_RGBA8888, the default color space.
However would you, please check whether the "emcoded" is correct in that location?

The note is in decoding section.

mijin85cho replied:
Hello, @jiyong2. Would you please check this comment again?

jiyong2 replied:
I'm sorry for the delay. The 'encoded with' is incorrect. It should be changed 'decoded with'.
I will replace the error.

jiyong2 commented:
The 'encoded with' has been repleaced to 'decoded with'
